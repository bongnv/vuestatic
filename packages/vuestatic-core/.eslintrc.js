const baseConfig = require("../../.eslintrc");

module.exports = {
  ...baseConfig,
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended",
  ],
  parser: "@typescript-eslint/parser",
  plugins: ["prettier", "@typescript-eslint"],
  ignorePatterns: ["/lib/", "__tests__"],
};
