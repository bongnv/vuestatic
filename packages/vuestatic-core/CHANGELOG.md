# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.1](https://github.com/bongnv/vuestatic/compare/@bongnv/vuestatic-core@0.1.0...@bongnv/vuestatic-core@0.1.1) (2020-06-13)

**Note:** Version bump only for package @bongnv/vuestatic-core





<a name="0.1.0"></a>
# [0.1.0](https://github.com/bongnv/vuestatic/compare/@bongnv/vuestatic-core@0.0.1...@bongnv/vuestatic-core@0.1.0) (2020-06-12)


### Features

* exclude node files from server bundle ([7e109d1](https://github.com/bongnv/vuestatic/commit/7e109d1))




# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.1 (2020-06-12)


### Bug Fixes

* fix analyze command ([95794e8](https://github.com/bongnv/vuestatic/commit/95794e823b5251cf4986df39398f129b199f40cc))
