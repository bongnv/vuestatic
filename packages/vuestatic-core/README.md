# `@bongnv/vuestatic-core`

> TODO: description

## Usage

```
const vuestaticCorePlugin = require('@bongnv/vuestatic-core');

// TODO: DEMONSTRATE API
```
