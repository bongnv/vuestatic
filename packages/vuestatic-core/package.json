{
  "name": "@bongnv/vuestatic-core",
  "version": "0.0.0",
  "description": "A VueStatic plugin for core logic. It provides clientWebpackConfig & serverWebpackConfig.",
  "author": "Bong Nguyen <van.bong@grabtaxi.com>",
  "homepage": "https://github.com/bongnv/vuestatic#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib",
    "vue-app"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bongnv/vuestatic.git"
  },
  "scripts": {
    "clean": "rimraf -rf ./lib",
    "tsc": "tsc",
    "watch": "tsc -w",
    "prepublishOnly": "npm run clean && npm run tsc",
    "lint": "eslint --fix --ext=js,jsx,ts --cache .",
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/bongnv/vuestatic/issues"
  },
  "dependencies": {
    "@babel/core": "^7.10.2",
    "@bongnv/vue-ssr-server-webpack-plugin": "^0.3.1",
    "babel-loader": "^8.1.0",
    "clean-webpack-plugin": "^3.0.0",
    "css-loader": "^3.5.3",
    "mini-css-extract-plugin": "^0.9.0",
    "postcss": "^7.0.32",
    "postcss-loader": "^3.0.0",
    "val-loader": "^2.1.1",
    "vue-loader": "^15.9.2",
    "vue-router": "^3.3.2",
    "vue-server-renderer": "^2.6.11",
    "vue-template-compiler": "^2.6.11",
    "webpack-chain": "^6.4.0"
  },
  "devDependencies": {
    "@types/mini-css-extract-plugin": "^0.9.1",
    "@types/webpack-chain": "^5.2.0"
  }
}
