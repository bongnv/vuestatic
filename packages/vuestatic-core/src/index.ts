export { BundleClientPlugin } from "./BundleClientPlugin";
export { BundleServerPlugin } from "./BundleServerPlugin";
export { Execution, ExecutionConfig } from "./Execution";
