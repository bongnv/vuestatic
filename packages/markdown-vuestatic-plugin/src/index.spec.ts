import { MarkdownVueStaticPlugin } from "./index";
import { Execution } from "@bongnv/vuestatic-core";

test("MarkdownVueStaticPlugin should create build config", async () => {
  const plugin = new MarkdownVueStaticPlugin();
  expect(plugin.apply).toBeTruthy();

  const e = new Execution({});
  plugin.apply(e);
  await e.commands.for("build").promise(e);
  await e.steps.for("config").promise(e);
  expect(e.config.serverWebpackConfig.toConfig().module?.rules).toHaveLength(3);
  expect(e.config.serverWebpackConfig.toConfig().resolve?.alias).toHaveProperty(
    "@content",
  );
  expect(e.config.serverWebpackConfig.toConfig().resolve?.alias).toHaveProperty(
    "@vuestatic/static-props",
  );
});

test("MarkdownVueStaticPlugin should create dev config", async () => {
  const plugin = new MarkdownVueStaticPlugin();
  expect(plugin.apply).toBeTruthy();

  const e = new Execution({});
  plugin.apply(e);
  await e.commands.for("dev").promise(e);
  await e.steps.for("config").promise(e);
  expect(e.config.serverWebpackConfig.toConfig().module?.rules).toHaveLength(3);
  expect(e.config.serverWebpackConfig.toConfig().resolve?.alias).toHaveProperty(
    "@content",
  );
  expect(e.config.serverWebpackConfig.toConfig().resolve?.alias).toHaveProperty(
    "@vuestatic/static-props",
  );
});
