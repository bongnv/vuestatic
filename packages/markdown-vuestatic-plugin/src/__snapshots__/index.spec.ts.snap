// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownVueStaticPlugin should create build config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.md\\$/,
        "use": Array [
          Object {
            "loader": "@bongnv/markdown-loader",
            "options": Object {
              "plugins": Array [
                [Function],
              ],
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(jpg\\|jpeg\\|png\\|svg\\|webp\\|gif\\|ico\\)\\$/,
        "use": Array [
          Object {
            "loader": "image-trace-loader",
          },
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[name].[ext]",
              "outputPath": "../../dist/_assets/images",
              "publicPath": "/_assets/images",
            },
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {
              "disable": true,
            },
          },
        ],
      },
    ],
  },
  "resolve": Object {
    "alias": Object {
      "@content": "/content",
    },
  },
}
`;

exports[`MarkdownVueStaticPlugin should create dev config 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.md\\$/,
        "use": Array [
          Object {
            "loader": "@bongnv/markdown-loader",
            "options": Object {
              "plugins": Array [
                [Function],
              ],
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.\\(jpg\\|jpeg\\|png\\|svg\\|webp\\|gif\\|ico\\)\\$/,
        "use": Array [
          Object {
            "loader": "image-trace-loader",
          },
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[name].[ext]",
              "outputPath": "static/_assets/images",
              "publicPath": "/_assets/images",
            },
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {
              "disable": true,
            },
          },
        ],
      },
    ],
  },
  "resolve": Object {
    "alias": Object {
      "@content": "/content",
    },
  },
}
`;
